client/src/components/StreamField/blocks/BaseSequenceBlock.js

```diff
@@ -230,17 +242,37 @@ export class BaseSequenceChild extends EventEmitter {
 
     // Inform the comment app that the content path of this block is no longer valid
     // This will hide any comments that were previously on the block
-    const contentPath = window.comments?.getContentPath(this.element);
+    const contentPath = this.getContentPath();
     if (contentPath && window.comments.commentApp) {
       window.comments.commentApp.invalidateContentPath(contentPath);
     }
   }
 
+  getContentPath() {
+    return window.comments?.getContentPath(this.element);
+  }
+
   enableDuplication() {
     this.emit('enableDuplication');
+    if (this.block && this.block.setCapabilityOptions) {
+      this.block.setCapabilityOptions('duplicate', { enabled: true });
```

Comment:
I'd be slightly happier if there was an 'onEnableDuplicate' event exposed in the capabilities dict that the child block/widget could subscribe to, rather than it being a call from the parent to the child - it feels like this is a more tightly-coupled "parent telling the child what to do" mechanism, rather than "here are some available capabilities, use them if you like". Not a blocker though, - this is something we can refine for 4.0.

Reply:
I can see that. The reason I switched to calling this function instead of events was it sounded like - with things like the new block insertion UI - it would often be useful to update other options within the capability, other than just enabled/disabled (which could be done by calling the event with arguments, just seemed a bit less clear).

I also wasn't sure how to implement this kind of event subscription without passing a reference to the parent to the child in order to subscribe to the emitter - I may be missing something obvious here. Do you have any thoughts on this?